classes/item.rb

```diff
@@ -2,12 +2,13 @@
 
 class Item
   attr_reader :id, :archived
-  attr_accessor :publish_date
+  attr_accessor :publish_date, :author
 
-  def initialize(publish_date, archived)
+  def initialize(publish_date, archived, author)
     @id = rand(1..100)
     @publish_date = Date.parse(publish_date)
     @archived = archived
+    @author = author
```

Comment:
I like the way you neatly presented your code, but based on the requirement, the author class should add self as a property of the item object **(by using the correct setter from the item object)**

Thus, consider removing the author from the initialize parameter in the item class and rather set it to nil by default within the initialize block. 

Create a setter 
`def author=(author) 
       @author = author 
       author.items.push(self) unless author.items.include?(self) 
end`

This will perform the assignment of the author to the item class and also correspond to the requirement of using the required setter to update the @author instance variable within the item class